packages/data-stores/src/site/actions.ts

```diff
@@ -95,25 +95,36 @@ export function createActions( clientCreds: WpcomClientCredentials ) {
 		type: 'RESET_RECEIVE_NEW_SITE_FAILED' as const,
 	} );
 
-	const launchSiteStart = ( siteId: number ) => ( {
-		type: 'LAUNCH_SITE_START' as const,
+	const launchSiteIdle = ( siteId: number ) => ( {
+		type: 'LAUNCH_SITE_IDLE' as const,
 		siteId,
 	} );
 
-	const launchSiteComplete = ( siteId: number ) => ( {
-		type: 'LAUNCH_SITE_COMPLETE' as const,
+	const launchSiteSuccess = ( siteId: number ) => ( {
+		type: 'LAUNCH_SITE_SUCCESS' as const,
 		siteId,
 	} );
 
+	const launchSiteFailure = ( siteId: number, error: string ) => ( {
+		type: 'LAUNCH_SITE_FAILURE' as const,
+		siteId,
+		error,
+	} );
+
 	function* launchSite( siteId: number ) {
-		yield launchSiteStart( siteId );
-		yield wpcomRequest( {
-			path: `/sites/${ siteId }/launch`,
-			apiVersion: '1.1',
-			method: 'post',
-		} );
-		yield launchSiteComplete( siteId );
-		return true;
+		yield launchSiteIdle( siteId );
+		try {
+			yield wpcomRequest( {
+				path: `/sites/${ siteId }/launch`,
+				apiVersion: '1.1',
+				method: 'post',
+			} );
+			yield launchSiteSuccess( siteId );
+			return true;
+		} catch ( error ) {
+			yield launchSiteFailure( siteId, error );
+			return false;
```

Comment:
Is it necessary to `return` a boolean? Do we use the return value anywhere?

Reply:
We never use the boolean anywhere so I removed it.